nuget/lib/dependabot/nuget/file_updater.rb

```diff
@@ -174,6 +186,34 @@ def updated_declaration(old_declaration, previous_req, requirement)
           requirement.fetch(:requirement)
         )
       end
+
+      def update_lock_files(files)
+        files = files.dup
+
+        lock_files = files.select { |f| lock_file?(f) }
+        lock_files.each do |lock_file|
+          project_file = files.find { |f| project_file?(f) && File.dirname(f.name) == File.dirname(lock_file.name) }
+          next if project_file.nil?
```

Comment:
Can you explain this check? 

It seems like there can be more than one lockfile (`packages.lock.json`) per project?

Reply:
In general there's a lockfile for every project in a solution, and transitive dependencies are represented in each lockfile.

So for example, you could have a structure like the following:
* dirs.sln (solution file, contains refs to projA & projB)
* projA/projA.csproj (proj file, has nuget reference to nugetX)
* projB/projB.csproj (proj file, has project reference to projA)

If dependabot runs for the solution file (dirs.sln), and decides to bump nugetX, you would expect to see:
* projA/packages.lock.json has a direct dependency, and is updated
* projB/packages.lock.json has a transitive dependency, and is also updated

Here's a concrete example of what this looks like in my project:
* Bicep.sln [references Bicep.Core & Bicep.Cli](https://github.com/Azure/bicep/blob/e620efb6c8fa43f4f180938177e2db4777330ee0/Bicep.sln#L6-L11)
* Bicep.Core has a [direct dependency on SharpYaml](https://github.com/Azure/bicep/blob/e620efb6c8fa43f4f180938177e2db4777330ee0/src/Bicep.Core/Bicep.Core.csproj#L37)
* Bicep.Cli has a [project dependency on Bicep.Core](https://github.com/Azure/bicep/blob/e620efb6c8fa43f4f180938177e2db4777330ee0/src/Bicep.Cli/Bicep.Cli.csproj#L25C67-L25C67) (and thus a transitive dependency on SharpYaml)
* Both of these lockfiles should be updated if SharpYaml is upgraded:
    * [Bicep.Core](https://github.com/Azure/bicep/blob/e620efb6c8fa43f4f180938177e2db4777330ee0/src/Bicep.Core/packages.lock.json#L207-L212)
    * [Bicep.Cli](https://github.com/Azure/bicep/blob/e620efb6c8fa43f4f180938177e2db4777330ee0/src/Bicep.Cli/packages.lock.json#L675-L679)